pkg/operator/status.go

```diff
@@ -309,6 +317,24 @@ func (c *Controller) syncStatus(cr *imageregistryv1.Config, deploy *appsapi.Depl
 	} else if cr.Spec.ManagementState == operatorapiv1.Removed {
 		operatorDegraded.Message = "The registry is removed"
 		operatorDegraded.Reason = "Removed"
+	} else if !isDeploymentStatusAvailable(deploy) {
+		if c.deployUnavailableSince.IsZero() {
+			c.deployUnavailableSince = time.Now()
+		}
+		if time.Since(c.deployUnavailableSince) > time.Minute {
+			operatorDegraded.Status = operatorapiv1.ConditionTrue
+			operatorDegraded.Message = "The deployment does not have available replicas"
+			operatorDegraded.Reason = "NoReplicasAvailable"
+		}
+	} else if !isDeploymentStatusComplete(deploy) {
+		if c.deployIncompleteSince.IsZero() {
+			c.deployIncompleteSince = time.Now()
+		}
+		if time.Since(c.deployIncompleteSince) > time.Minute {
+			operatorDegraded.Status = operatorapiv1.ConditionTrue
+			operatorDegraded.Message = "The registry has minimum availability"
+			operatorDegraded.Reason = "MinimumAvailability"
```

Comment:
This reason/message sounds positive.  If we are going to set Degraded=True, we should say something negative to motivate the unhappy status.  Something like `Registry deployment has timed out progressing`.  But... can we lean on the Deployment controller for this and look at `ProgressDeadlineExceeded`?  See my rotted-due-to-lack-of-interest openshift/library-go#575 for one possible implementation.

Reply:
Done, PTAL.